thorlcr/activities/lookupjoin/thlookupjoinslave.cpp

```diff
@@ -923,17 +938,17 @@ class CInMemJoinBase : public CSlaveActivity, public CThorDataLink, public CAllO
                         throw MakeActivityException(this, 0, "Out of memory: Unable to add any more rows to RHS");
 
                     rightSerializer->serialize(mbser, (const byte *)row.get());
-                    if (mb.length() >= MAX_SEND_SIZE || broadcaster.isStopping())
+                    if (mb.length() >= MAX_SEND_SIZE || broadcaster.stopRequested())
```

Comment:
Yes that's correct and that was the problem.
When CBroadcaster.start() called setStopping(), it was because the activity was stopping before having read anything. That caused the broadcaster to prematurely stop and spuriously signal to other nodes that it had spilt. Spilt state was wrong and the other slaves may well still be reading and therefore require the complete broadcast. Ultimately the spurious communicated spilt state caused a deadlock, but the stopping of the broadcast was wrong.